cordformation/src/main/kotlin/net/corda/plugins/Signing.kt

```diff
@@ -0,0 +1,65 @@
+package net.corda.plugins
+
+import org.gradle.api.tasks.Input
+import java.nio.file.Path
+
+//contains union of properties for ANT tasks "genkey" and "signjar" as most of them are shared by both tasks
+class Signing {
+
+    var defaultKeyStoreFileName = "jarSignKeystore.jks"
+    var defaultStoretype = "jks"
+    var defaultStorepass = "secret1!"
+    var defaultAlias = "cordapp-signer"
+    var defaultDname = "OU=Dummy Cordapp Distributor, O=Corda, L=London, C=GB"
+
+    @get:Input
+    var all: Boolean = false
+        private set
+    @get:Input
+    var enabled: Boolean = true
+        private set
+    @get:Input
+    var generateKeystore: Boolean = true
+        private set
+
+    private var opts = mutableMapOf("storetype" to defaultStoretype,
+            "alias" to defaultAlias,
+            "storepass" to defaultStorepass,
+            "dname" to defaultDname,
+            "keyalg" to "RSA") //required by Corda
+        set(value) {
+            opts.putAll(value)
+        }
+
+    fun all(value: Boolean) {
+        all = value
+    }
+
+    fun enabled(value: Boolean) {
+        enabled = value
+    }
+
+    fun generateKeystore(value: Boolean) {
+        generateKeystore = value
+    }
+
+    fun genKeyTaskOptions(baseDirectory: Path): Map<String, String> {
```

Comment:
We should keep anything that's not intended for the DSL off of the DSL classes to avoid changing the DSL lateer.

Reply:
(I know other classes don't, but that's changing)